### ISSUE TITLE
tf.GradientTape not working properly.

### ISSUE THREAD

**Issue by uoyssim** [tensorflow/tensorflow#51795]
Recently I made new network & trained it with tf.Gradient.

But losses didn't decrease.

So I tested in different examples in [tensorflow tutorial](https://www.tensorflow.org/tutorials/images/cnn?hl=en)

When I tested, I used tensorflow turorial code except training part.

Please let me know if there is any wrong usage.

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution: Linux Ubuntu 20.04 
- TensorFlow installed from (source or binary): pip install 
- TensorFlow version (use command below): 2.4.0 / 2.6.0
- Python version: Python 3.8
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 11.4 
- GPU model and memory: GeForce 1080 Ti * 2

v2.6.0-rc2-32-g919f693420e 2.6.0

**Describe the current behavior**
![Screenshot from 2021-09-02 19-45-42](https://user-images.githubusercontent.com/18378446/131830787-5992bee1-7967-403b-b75e-81c74c0cc27e.png)

**Describe the expected behavior**
![Screenshot from 2021-09-02 19-46-27](https://user-images.githubusercontent.com/18378446/131830878-9d5a2349-4cff-4bd5-bced-48138beefb60.png)

**Comment by uoyssim** [comment]
Thanks for the quick response :)

```python
from tensorflow.keras import datasets, layers, models

(train_images, train_labels), (test_images, test_labels) = datasets.cifar10.load_data()
train_images, test_images = train_images / 255.0, test_images / 255.0

class_names = ['airplane', 'automobile', 'bird', 'cat', 'deer',
               'dog', 'frog', 'horse', 'ship', 'truck']

seq_model = models.Sequential()
seq_model.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(32, 32, 3)))
seq_model.add(layers.MaxPooling2D((2, 2)))
seq_model.add(layers.Conv2D(64, (3, 3), activation='relu'))
seq_model.add(layers.MaxPooling2D((2, 2)))
seq_model.add(layers.Conv2D(64, (3, 3), activation='relu'))
seq_model.add(layers.Flatten())
seq_model.add(layers.Dense(64, activation='relu'))
seq_model.add(layers.Dense(10))

train_loss_results = []
train_accuracy_results = []

num_epochs = 10
total_size = len(train_images)
batch_size = 128
n_steps = int(total_size / batch_size)

for epoch in range(num_epochs):
    epoch_loss_avg = tf.keras.metrics.Mean()
    epoch_accuracy = tf.keras.metrics.SparseCategoricalAccuracy()

    np.random.seed(i)
    np.random.shuffle(train_images)
    np.random.seed(i)
    np.random.shuffle(train_labels)
    
    for i in range(n_steps):
        upper = min((i+1) * batch_size, total_size)
        x = train_images[i*batch_size:upper]
        y = train_labels[i*batch_size:upper]
        
        with tf.GradientTape() as tape:
            preds = seq_model(x, training=True)
            loss = loss_object(y, preds)
        
        grads = tape.gradient(loss, seq_model.trainable_variables)
        optimizer.apply_gradients(zip(grads, seq_model.trainable_variables))

        epoch_loss_avg(loss) 
        epoch_accuracy(y,seq_model(x))

    train_loss_results.append(epoch_loss_avg.result())
    train_accuracy_results.append(epoch_accuracy.result())

    print("Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(epoch+1,
                                                            epoch_loss_avg.result(),
                                                            epoch_accuracy.result()))
```

**Comment by sushreebarsa** [comment]
@uoyssim Thank you for the response! 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by sushreebarsa** [comment]
@uoyssim Could you please close this ticket if you have posted the issue in Keras repo ?Thank you!

**Comment by uoyssim** [comment]
[Keras Issues](https://github.com/keras-team/keras/issues/15306) is closed.

It is my mistake. 

When I use tf.keras.losses.SparseCategoricalCrossentropy,I missed 'from_logits=True'

Thank you for your helps!
